### ISSUE TITLE
Note keeping and notebooks + notes folder

### ISSUE THREAD

**Issue by leewujung** [uw-echospace/duty-cycle-investigation#1]
Hey @aditya-uw: This is something I thought about this morning while going through some of Caesar's recent work in the `hake-labels` repo.

I think the below would be useful for you in the subsampling project too so want to just add it here also:
- Set up a folder in this repo and date your notebooks, so that there are records of the iterations
  - You already have a `notebooks` folder, which I assume you keep the latest version of things there. Maybe you could add another folder like `notebook_daily` to go with the detailed notes
  - You can still keep revising the same notebook, just make a copy and update the date in the filename
  - Think of these as the computational equivalent of the lab notebooks people use in their experimental work
- Set up a `notes` folder in this repo, and put detailed notes daily on what you have tried and worked/not worked, anything to follow up later
  - I have found keeping notes like this to be incredibly useful when trying to figure out why a certain decision was made, what problems were encountered, and keep track of what still needs to be done
  - See example [here](https://github.com/leewujung/pp_expt) of my notes on my analysis of the porpoise spheroid discrimination expt data), and [here](https://github.com/leewujung/pp_expt_data_analysis/tree/main/notebooks) for my daily-ish analysis notebooks
  - I know you've been writing the daily update slides (and they are excellent!) -- you can continue to do that if you would like, or just switch to write the detailed notes here.

**Comment by leewujung** [comment]
I also thought for the notes you can just do direct push since I don't have to review them. For the notebooks you can be direct push or PR depending on whether you think it is something we should discuss -- perhaps you could pick one approach, and we see how that goes and change if needed.

**Comment by leewujung** [comment]
BTW, I had 2 repos for the analysis notebooks and the notes, but that was because of some historical reasons (the `pp_expts` repo got wayyy to messy due to some change of plan), and I usually put the notes and notebooks in the same repo.

**Comment by aditya-uw** [comment]
I see! Thanks for the suggestion @leewujung! Quick question about the daily notebooks:
1) Is it better to make a copy of the complete notebook that I made updates to? Or would it be better to just make a notebook around the updates I worked on for that date? I'm thinking about the times where I have a notebook I'm working in but I only changed about 20% of it because the rest 80% is stuff I have tested before that relate to the notebook's purpose.
   - I also think this may have been what you did for your analysis notebooks [here](https://github.com/leewujung/pp_expt_data_analysis/tree/main/notebooks) because they all seem to only contain whatever their titles suggest?

**Comment by leewujung** [comment]
Yeah you are right, that's the scenario I haven't found a good way to address -- I often forgot to change the notebook title!!
The pattern that I found I've been following lately is to copy the notebook and immediately change the filename with the new date, and if I ended up making a lot of changes to the new notebook, I summarize the changes in the header section immediately below the notebook title, something like 
```
This notebook was based on a previous notebook SOME_NOTEBOOK_FILENAME, but with the following modification:
- change1
- change2
- change3 ...
```
